sql/mysqld.cc

```diff
@@ -2430,7 +2430,7 @@ static MYSQL_SOCKET activate_tcp_port(uint port)
   if (ret < 0)
   {
     char buff[100];
-    sprintf(buff, "Can't start server: Bind on TCP/IP port. Got error: %d",
+    snprintf(buff, sizeof(buff), "Can't start server: Bind on TCP/IP port. Got error: %d",
```

Comment:
obviously fits

Reply:
The point of these changes is very specifically to make them _**even where it's “obvious” that there's no overrun possibility**_ in the current code.

1. This will reduce the cognitive load and potential for oversights and errors _in the future_: future reviewers won't have to manually (re)review so many occurrences of (unbounded) `sprintf` or `strcpy` to (re)confirm that they're “obviously” correct.
2. Compilers can optimize `snprintf` into `sprintf`, `strncpy` into `strcpy`, etc, when they can verify at compile time that no truncation is possible.
3. The vast majority of these changes are in error paths / slow paths.

    If any of them is in a truly hot path where it was plausibly performance-limiting and might require hand-optimization, we could revisit it individually.